### PR TITLE
Generic worker

### PR DESCRIPTION
**Pull request by dividehex**

### CODE REVIEW

File: modules/generic_worker/manifests/control_bug.pp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            group  => $::root_group;
+
+        '/usr/local/share/generic-worker/bugzilla-utils.sh':
+            ensure  => present,
+            content => template('generic_worker/bugzilla-utils.sh.erb'),
+            mode    => '0755',
+            owner   => $::root_user,
+            group   => $::root_group;
+    }
+}
```
Comment (dragoscrisan) [inline modules/generic_worker/manifests/control_bug.pp]: This class was modified. See https://github.com/mozilla-releng/build-puppet/pull/409/files

File: modules/generic_worker/manifests/init.pp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+            httpd::config { 'proxy.conf':
+                content => template('generic_worker/proxy-httpd.conf.erb'),
+            }
+        }
+        default: {
+            fail("${module_name} is not supported on ${::operatingsystem}")
+        }
+    }
+}
```
Comment (dragoscrisan) [inline modules/generic_worker/manifests/init.pp]: This class was modified. See https://github.com/mozilla-releng/build-puppet/pull/409/files

File: modules/generic_worker/templates/bugzilla-utils.sh.erb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+         "body": "The worker is now in working state",
+         "is_private": false
+      }
+   }
+EOF
+)
+   echo "$JSON" > $json_file
+
+  curl -v -X PUT -H Content-Type:application/json -H Accept:application/json -d @$json_file https://bugzilla.mozilla.org/rest/bug/$bug_id
+}
```
Comment (dragoscrisan) [inline modules/generic_worker/templates/bugzilla-utils.sh.erb]: This file will not be used in the future. See This class was modified. See https://github.com/mozilla-releng/build-puppet/pull/409/files

File: modules/generic_worker/templates/generic-worker.config.erb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  "workerGroup": "<%= @worker_group %>",
+  "workerId": "<%= @hostname %>",
+  "workerType": "<%= @worker_type %>",
+  "workerTypeMetadata": {
+    "machine-setup": {
+      "config": "https://github.com/mozilla-platform-ops/ronin_puppet/raw/master/modules/generic_worker/templates/generic-worker.config.erb",
+      "docs": "https://wiki.mozilla.org/ReleaseEngineering/PuppetAgain/Modules/generic_worker"
+    }
+  }
+}
```
Comment (dragoscrisan) [inline modules/generic_worker/templates/generic-worker.config.erb]: Looks good

File: modules/generic_worker/templates/run-generic-worker.sh.erb [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        /bin/rm -rf <%= @user_homedir %>/bug_id
+    fi
+    # Now reboot the machine immediately (time costs money)
+    <%= @reboot_command %>
+    # Sleep to prevent this script from terminating naturally, and launchd restarting
+    # it. Instead, the shutdown should cause this script to terminate (so it won't
+    # really sleep for 2 mins). If shutdown doesn't kick in within 2 mins, it is sane
+    # for this script to exit, and allow launchd to fire up the worker again.
+    /bin/sleep 120
+fi
```
Comment (dragoscrisan) [inline modules/generic_worker/templates/run-generic-worker.sh.erb]: This file was modified. See https://github.com/mozilla-releng/build-puppet/pull/409/files

File: modules/roles_profiles/manifests/profiles/generic_worker.pp [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+            $quarantine_access_token  = lookup('quarantine_access_token')
+
+            class { 'generic_worker':
+                taskcluster_client_id     => $taskcluster_client_id,
+                taskcluster_access_token  => $taskcluster_access_token,
+                livelog_secret            => $livelog_secret,
+                worker_group              => regsubst($facts['networking']['fqdn'], '.*\.releng\.(.+)\.mozilla\..*', '\1'),
+                quarantine_client_id      => $quarantine_client_id,
+                quarantine_access_token   => $quarantine_access_token,
+                generic_worker_version    => 'v13.0.4',
```
Comment (dragoscrisan) [inline modules/roles_profiles/manifests/profiles/generic_worker.pp]: Now, we use generic-worker v13.0.3

File: modules/generic_worker/manifests/control_bug.pp (hunk 1 above)
Comment (dividehex) [reply modules/generic_worker/manifests/control_bug.pp]: I'll merge the new work here when that pull request is fully reviewed and merged.  For right, now i'm going to import what already exists.

File: modules/roles_profiles/manifests/profiles/generic_worker.pp (hunk 6 above)
Comment (dividehex) [reply modules/roles_profiles/manifests/profiles/generic_worker.pp]: Is there a reason we are not using the latest version of generic worker?

**Pull request merged**